### PR TITLE
UI Part for Filter & Search Utility

### PR DESCRIPTION
**Pull request by harshit-sharma-gits**
I have made the UI Part of the Filter & Search Issue #119, and it is looking like [this](https://drive.google.com/file/d/1DefTypceAXoKBFU-hRbXs2dg_Ewnyxxp/view?usp=sharing)

Changes Summary:
* Added Profile MenuField and Category MenuField
(Currently, I have initialized them with random values)

* Added Search TextControl

Also, all the clickables present in the UI, currently sends a tempMessage that I created, and it is not functional yet

### CODE REVIEW

**Comment by harshit-sharma-gits** [comment]
> If after using the app a bit, it turns out more fine-grained filtering needs a category pop-up menu, there's no problem adding it. But to start out needlessly complicated...
>
> Let's stick to one feature at a time, the filtering, and leave "profiles" until this one is working.

Okay, so should I remove the unnecessary things, that is, Profile and Categories MenuFiled.
And leave "Search" Text Control and the clear button there?

Well I don't think having the Clear button would be a nice thing, as only a Search Box is present, and if we implement a category popup menu in future, then we may have the Clear button as well.

**Comment by humdingerb** [comment]
Yes, as I wrote in #119, just the BTextControl labelled "Filter:". No "Clear" button. It'd be nice to capture ESC to clear the filter. But that can come later.

File: src/SidePanelView.cpp [hunk 1]
```diff
@@ -68,9 +73,45 @@ SidePanelView::SidePanelView()
 		BSize(font.StringWidth(B_TRANSLATE_COMMENT(
 				"September XXXX", "Choose the longest month name")),
 			height));
-
+
```
Comment (nielx) [inline src/SidePanelView.cpp:76]: Extra whitespace

**Comment by nielx** [comment]
I added some comments throughout the code.

Two general remarks about PR hygiene: 

1. Your changes are spread over three commits. While I can imagine you make these commits internally while working on it and wanting to keep the changes, when you are going to upstream these changes you generally want to clean them up, i.e. squash them into one commit. You also want to give them clear, concise commit messages about the change. 'Get Started' is not one of those.
2. You are adding the Calendar.pld file. It is fine to use this locally, but it should not be added remotely. The official build method is to use the Makefile, so we do not need additional IDE files for this. Please remove this from the commit.

Furthermore, I agree with @humdingerb about keeping it simple. As we discussed on our call on Sunday, let's start with text filtering. We can add the other things later.

**Review by nielx — CHANGES_REQUESTED** [review]
See individual comments and general comment.

File: src/SidePanelView.cpp [hunk 2]
```diff
@@ -11,6 +11,9 @@
 #include <LayoutBuilder.h>
 #include <LocaleRoster.h>
 #include <StringView.h>
+#include <MenuField.h>
```
Comment (humdingerb) [inline src/SidePanelView.cpp:14]: Alphasort includes.

File: src/SidePanelView.cpp [hunk 3]
```diff
@@ -48,6 +51,8 @@ SidePanelView::SidePanelView()
 
 	fYearLabel = new BStringView("year", "");
 	fMonthLabel = new BStringView("month", "");
+	fFilterSearchHeading = new BStringView(BRect(10,10,11,11), "filterSearchHeading", "Filter & Search");
```
Comment (humdingerb) [inline src/SidePanelView.cpp:54]: You'd want to use the layout aware constructor without the first BRect parameter.
But: We don't need this heading anyway, the label "Filter:" of the BTextView is clue enough.

**Review by humdingerb — COMMENTED** [review]
I suppose some comments a FYI only, since you'll remove some of the code when updating the PR anyway.

**Comment by humdingerb** [comment]
> Two general remarks about PR hygiene:

WRT working on PRs in general, I recommend using "feature branches". There's a nice intro to it at https://github.com/haikuports/haikuports/wiki/DevelopmentModel
That way, you can easily leave your work in one branch for a bit and do some experimenting in another.

It's also a good idea to separate your work into distinct commits, e.g. do style fixes in one commit, GUI work in another and backend work in yet another. Before creating your PR, do a "git -i rebase" to move those commits around and squash the commits of a similar area.

**Comment by humdingerb** [comment]
Anything new, @harshit-sharma-gits ?
It's been very quiet for too long...

**Comment by harshit-sharma-gits** [comment]
Hello!
As mentioned, I have created a new branch, namely, filterFeature, for implementing the Filter Utility
And cleared the excess clutter from the main window, now the updated UI looks like this -

UI Part Refined

Now, the current goal for the filter is to capture the keywords, and search them in the events list for the particular tab (Day, Week, or Month) and display them accordingly.

**Comment by humdingerb** [comment]
You haven't pushed that branch to your repo, it appears. If you update this PR, we can keep track of your progress. Just adding a BTextControl to the window seems a bit scant for a week of full time work...
Do speak up when you're stuck somewhere and need some pointers.

**Comment by nielx** [comment]
Superseded by PR #121

**Pull request closed without merging**